smartdashboard/views.py

```diff
@@ -341,3 +345,467 @@ def __init__(
         self.app_view = app_view
         self.ens_view = ens_view
         self.orc_view = orc_view
+
+    def update(self) -> None:
+        """Update all views within the OverviewView"""
+        self.exp_view.update()
+        self.app_view.update()
+        self.ens_view.update()
+        self.orc_view.update()
+
+
+@dataclass(frozen=True)
+class Files:
+    graph_file: str
+    table_file: str
+
+    def is_valid(self) -> bool:
+        """Check that paths are not empty strings and exist"""
+        if any(file_path == "" for file_path in (self.graph_file, self.table_file)):
+            return False
+
+        if not all(
+            os.path.exists(file_path)
+            for file_path in (self.graph_file, self.table_file)
+        ):
+            return False
+
+        return True
+
+
+class DualView(ViewBase):
+    """Base class for Telemetry Views that have table and chart elements"""
+
+    def __init__(
+        self,
+        shard: t.Optional[Shard],
+        table_element: DeltaGenerator,
+        graph_element: DeltaGenerator,
+        export_button: DeltaGenerator,
+    ):
+        self.shard = shard
+        self.table_element = table_element
+        self.graph_element = graph_element
+        self.export_button = export_button
+        self.window_size = 10000
+        self.telemetry_df = pd.DataFrame(columns=self.columns)
+        self.sampling = False
+        self.chart: t.Optional[alt.Chart] = None
+
+        if self.telemetry:
+            self.telemetry_df = self.set_telemetry_df()
+            self.timestamp_min = self.telemetry_df["timestamp"].min()
```

Comment:
It looks like this attr is only instanced if the `self.telemetry:` branch is taken. Can we add a sensible default just in case `self.telemetry == False`?

```py
if self.telemetry:
    ...
    self.telemetry_df = self.set_telemetry_df()
    ...
else:
    self.timestamp_min = ...  # <-- TODO: some sensible default here
```

Reply:
I added `self.timestamp_min = 0` when initializing the other telemetry info. Let me know if that's bad lol

Reply:
That seems like a sensible default to me!